File: 2-6/2-6-Java/FlyTest.java
/*
 * Program to use Airplane and Jet classes.
 *
 * @author  Jakub Malhotra
 * @version 1.0
 * @since   2024-10-29
 */

/**
 * This is the FlyTest program that demonstrates
 * the functionality of the Airplane and Jet classes.
 */
public class FlyTest {

    /**
     * The main() method to execute the program.
     *
     * @param args No command-line arguments used
     */
    public static void main(String[] args) {
        Airplane biplane = new Airplane();
        biplane.setSpeed(212);
        System.out.println(biplane.getSpeed());

        Jet boeing = new Jet();
        boeing.setSpeed(422);
        System.out.println(boeing.getSpeed());

        int x = 0;
        while (x < 4) {
            boeing.accelerate();
            System.out.println(boeing.getSpeed());

            if (boeing.getSpeed() > 5000) {
                biplane.setSpeed(biplane.getSpeed() * 2);
            } else {
                boeing.accelerate();
            }
            x++;
        }
        System.out.println(biplane.getSpeed());
    }
}

File: 2-6/2-6-TS/FlyTest.ts
/**
 * This is the main code that runs
 * with the Plane and Jet classes.
 * By:      Jakub Malhotra
 * Version: 1.0
 * Since:   2024-10-29
 */

import Plane from './Airplane.ts';
import Jet from './Jet.ts';

const biplane = new Plane();
biplane.setSpeed(212);
console.log(biplane.getSpeed);

const boeing = new Jet();
boeing.setSpeed(422);
console.log(boeing.getSpeed);

let x = 0;
while (x < 4) {
    boeing.accelerate();
    console.log(boeing.getSpeed);
    if (boeing.getSpeed > 5000) {
        biplane.setSpeed(biplane.getSpeed * 2);
    } else {
        boeing.accelerate();
    }
    x++;
}

console.log(biplane.getSpeed);
